### DISCUSSION TITLE
[riverpod_lint] Correctly adapt to *scoped_providers_should_specify_dependencies* in testing environments

### DISCUSSION THREAD

**Discussion by timokz** [rrousselGit/riverpod#2913] [Q&A] [answered]
So I've been using [alchemist](https://github.com/Betterment/alchemist) for golden testing, after adding riverpod lints, I'm puzzled by the warning of: 
"Providers which are overridden in a non-root ProviderContainer/ProviderScope should specify dependencies.dart(scoped_providers_should_specify_dependencies)" 

I've extracted and simplified a POC from my codebase to demonstrate what I'm encountering: 
```dart
void main() {
  group('OBDButton Golden Tests', () {
    goldenTest(
      'renders correctly',
      fileName: 'obd_button',
      tags: ['golden'],
      builder: () => GoldenTestGroup(
        children: [
          GoldenTestScenario(
            name: 'with vehicle selected',
            child: ProviderScope(
              overrides: [
                obdButtonColorProvider.overrideWithValue(Colors.green),
// warning occurs here
                obdStateProvider.overrideWith(
                  (ref) => const Stream<OBDState>.empty(),
                ),
              ],
              child: const OBDButton(),
            ),
          ),
          GoldenTestScenario(
            name: 'without vehicle selected',
            child: ProviderScope(
              overrides: [
                obdButtonColorProvider.overrideWithValue(Colors.red),
              ],
              child: const OBDButton(),
            ),
          ),
        ],
      ),
    );
  });
}
```

I can see why custom lints says I'm in a non-root context here, but I'm not quite sure how to specify the dependencies of a simple generated StreamProvider like the one I'm using here. 
Thanks to everybody for reading this far and your contributions to Riverpod, looking forward to any advice.

**Comment by rrousselGit** [comment, marked as answer]
This is likely a false positive due to the unusual nature of your test. You can ignore the warning for now. I'll think about a way to handle this better.

**Reply by timokz** [reply]
Thanks for answering!
